test/controllers/mobileNumber/CaptureMobileNumberControllerSpec.scala

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.mobileNumber
+
+import assets.BaseTestConstants._
+import assets.CustomerInfoConstants.fullCustomerInfoModel
+import common.SessionKeys._
+import controllers.ControllerBaseSpec
+import mocks.MockVatSubscriptionService
+import models.errors.ErrorModel
+import org.jsoup.Jsoup
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{never, verify}
+import play.api.http.Status
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.errors.NotFoundView
+import views.html.mobileNumber.CaptureMobileNumberView
+
+import scala.concurrent.ExecutionContext
+
+class CaptureMobileNumberControllerSpec extends ControllerBaseSpec with MockVatSubscriptionService {
+
+  val controller = new CaptureMobileNumberController(
+    mockVatSubscriptionService,
+    mockErrorHandler,
+    inject[CaptureMobileNumberView],
+    inject[NotFoundView]
+  )
+
+  "Calling the show action" when {
+
+    "a user is enrolled with a valid enrolment" when {
+
+      "the user's current mobile is retrieved from session" should {
+
+        lazy val result = controller.show(requestWithValidationPhoneNumbers)
+
+        lazy val document = Jsoup.parse(bodyOf(result))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+
+        "prepopulate the form with the user's current mobile" in {
+          document.select("#mobileNumber").attr("value") shouldBe testValidationMobile
+        }
+
+        "not call the VatSubscription service" in {
+          verify(mockVatSubscriptionService, never())
+            .getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext])
+        }
+
+      }
+    }
+
+    "the previous form value is retrieved from session" should {
+
+      lazy val result = controller.show(requestWithAllContactNumbers)
```

Comment:
Can you make one of these specific to mobile please and make the existing one specific to landline? We wouldn't usually have this setup in a usual user journey.